plugins/contiv/plugin_impl_contiv.go

```diff
@@ -620,3 +623,100 @@ func appendIfMissing(slice []string, s string) []string {
 	}
 	return append(slice, s)
 }
+
+// getIPAMConfig populates the Config struct with the calculated subnets
+func getIPAMConfig(config *Config) *Config {
+	_, contivNetwork, _ := net.ParseCIDR(config.IPAMConfig.ContivCIDR)
+	maskSize, _ := contivNetwork.Mask.Size()
+	subnetPrefixLength := 23 - maskSize
+
+	podSubnetCIDR, _ := subnet(contivNetwork, 1, 0)
+	podNetworkPrefixLen := uint8(24)
+	vppHostSubnetCIDR, _ := subnet(contivNetwork, 1, 1)
+	vppHostNetworkPrefixLen := uint8(24)
+	nodeInterconnectCIDR, _ := subnet(contivNetwork, subnetPrefixLength, 125)
+	podIfIPCIDR, _ := subnet(contivNetwork, subnetPrefixLength, 126)
+	vxlanCIDR, _ := subnet(contivNetwork, subnetPrefixLength, 127)
+
+	config.IPAMConfig = ipam.Config{
+		PodIfIPCIDR:             podIfIPCIDR.String(),
+		PodSubnetCIDR:           podSubnetCIDR.String(),
+		PodNetworkPrefixLen:     podNetworkPrefixLen,
+		VPPHostSubnetCIDR:       vppHostSubnetCIDR.String(),
+		VPPHostNetworkPrefixLen: vppHostNetworkPrefixLen,
+		VxlanCIDR:               vxlanCIDR.String(),
+	}
+
+	if config.IPAMConfig.NodeInterconnectDHCP != true {
+		config.IPAMConfig.NodeInterconnectCIDR = nodeInterconnectCIDR.String()
+	}
+
+	return config
+}
+
+// subnet takes a CIDR range and creates a subnet from it
+// base: parent CIDR range
+// newBits: number of additional prefix bits
+// num: given network number.
+//
+// Example: 10.1.0.0/16, with additional 8 bits and a network number of 5
+// result = 10.3.5.0/24
```

Comment:
result should be `10.1.5.0/24` shoudln't it?

Reply:
yup typo